File: service/src/main/java/com/github/goldy1992/mp3player/service/library/MediaLibrary.java
package com.github.goldy1992.mp3player.service.library;

import android.net.Uri;
import android.support.v4.media.MediaBrowserCompat.MediaItem;

import androidx.annotation.NonNull;

import com.github.goldy1992.mp3player.commons.MediaItemType;
import com.github.goldy1992.mp3player.service.library.content.retriever.ContentResolverRetriever;

import java.util.List;
import java.util.Map;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class MediaLibrary {

    private boolean playlistRecursInSubDirectory = false;

    private Map<MediaItemType, ContentResolverRetriever> contentRetrieverMap;
    private ContentManager contentManager;
    private final String LOG_TAG = "MEDIA_LIBRARY";


    @Inject
    public MediaLibrary(ContentManager contentManager) {
        this.contentManager = contentManager;
    }

    public List<MediaItem> getChildren(@NonNull String parentId) {
        return contentManager.getChildren(parentId);
    }





    public Uri getMediaUriFromMediaId(String mediaId){
        throw new UnsupportedOperationException();
    }

}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/MediaLibrary.kt
package com.github.goldy1992.mp3player.service.library

import androidx.media3.common.MediaItem
import com.github.goldy1992.mp3player.commons.LogTagger
import dagger.hilt.android.scopes.ServiceScoped
import javax.inject.Inject

@ServiceScoped
class MediaLibrary @Inject constructor(private val contentManager: ContentManager) : LogTagger {

    fun getChildren(parentId: String): List<MediaItem?> {
        return contentManager.getChildren(parentId)
    }

    override fun logTag(): String {
        return "MEDIA_LIBRARY"
    }

}